### DISCUSSION TITLE
I made a blender plugin for pos and rot and qa about gear rotation

### DISCUSSION THREAD

**Discussion by peteblank** [google-deepmind/mujoco#1924] [General]
Hello, I made a blender plugin that gets the rotation and position of and exports it to the  mujoco format
https://github.com/HannahAnimatronic/mujoco_blender_plugin
The reason this isn't show and tell is cause I also have a question. I made two gears but I haven't been able to make it slow and smooth and I'm unsure if its going the full 360 degrees.
```
<?xml version="1.0"?>
<mujoco model="testing2">
  <default>
    <default class="visual">
      <geom group="2" type="mesh" contype="0" conaffinity="0"/>
    </default>
    <default class="collision">
      <geom group="3" type="mesh"/>
    </default>
  </default>
  
  <asset>
    <mesh file="Gear_1.obj"/>
    <mesh file="Gear_2.obj"/>
  </asset>
  
  <worldbody>
    <!-- Gear 1 with Joint -->
    <body name="Gear_1" zaxis="0 1 0">
      <geom type="mesh" mesh="Gear_1" class="visual" quat="1.0 0.0 0.0 0.0" pos="0.0 0.0 0.0" rgba="1.0 1.0 1.0 1.0"/>
      <geom type="mesh" mesh="Gear_1" class="collision" quat="1.0 0.0 0.0 0.0" pos="0.0 0.0 0.0"/>
      <joint name="Gear_1_joint" type="hinge" axis="0 0 1" pos="0 0 0" range="180 -180"/>
    </body>
    
    <!-- Gear 2 with Joint -->
    <body name="Gear_2" zaxis="0 1 0">
      <geom type="mesh" mesh="Gear_2" class="visual" quat="1.0 0.0 0.0 0.0" pos="16.320199966430664 20.766149520874023 0.458035945892334" rgba="1.0 1.0 1.0 1.0"/>
      <geom type="mesh" mesh="Gear_2" class="collision" quat="-1.0 0.0 0.0 0.0" pos="16.320199966430664 20.766149520874023 0.458035945892334"/>
      <joint name="Gear_2_joint" type="hinge" axis="0 0 1" pos="16.320199966430664 20.766149520874023 0.458035945892334"/>
    </body>
  </worldbody>

  <actuator>
    <motor name="Gear_1_motor" joint="Gear_1_joint" gear="1" ctrlrange="-0.05 0.05"/>
    <velocity name="Gear_1_velocity" joint="Gear_1_joint" kv="50"/>
  </actuator>
</mujoco>
```

https://github.com/user-attachments/assets/221e5df4-f64c-4b93-a7a8-1648ff9b8f7a

**Comment by bmaxdk** [comment]
joint range from 180 to -180 degrees represent a total of 360 degrees of rotation. However, the way it's specified could potentially cause issues depending on how the simulation interprets this range. this might lead to situations where the gear rotation appears to "snap" or "jump" when crossing the boundary between 180 and -180 degrees. Can you try whether it works with `"0 360"`?

Also your `ctrlrange="-0.05 0.05"` in the motor actuator may be too low if necessary, you may increase to `"-1 1"` for testing.

Hope your problem resolved

**Comment by peteblank** [comment]
Thank you. But this is strange. I tried adding texture to gear2 and saving it and now mujoco is not acting as intended...
![Screenshot (62)](https://github.com/user-attachments/assets/1e745a92-525b-4037-a77b-fdede5dde653)

**Reply by bmaxdk** [reply]
Make sure the texture you applied in Blender is compatible w/ the mujoco since mujoco supports a limited set of texture and material format. I seen the issue when the material property. You may want to debug without the texture or applied with simple texture. It also could be appeared with mesh after texturing. Try it. Hope everything works well
